### PR TITLE
Adds back recovery keys api's

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/12274

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [x] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [x] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- restore wallet that has user funds
- go to backup tab and make sure that you see backup keys

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 1]
```diff
@@ -3373,5 +3373,13 @@ void RewardsServiceImpl::StartProcess(StartProcessCallback callback) {
   StartLedger(std::move(callback));
 }
 
+void RewardsServiceImpl::GetWalletPassphrase(
+    const GetWalletPassphraseCallback& callback) {
+  if (!Connected()) {
+    return;
```
Comment (bbondy) [inline components/brave_rewards/browser/rewards_service_impl.cc:3379]: Do we need to call the callback before we return here? Are the callers expecting a response?

Comment (NejcZdovc) [reply components/brave_rewards/browser/rewards_service_impl.cc:3379]: it's not expecting, but let me add it for consistency

Comment (NejcZdovc) [reply components/brave_rewards/browser/rewards_service_impl.cc:3379]: done

**Review by bbondy — APPROVED** [review]

**Review by tmancey — APPROVED** [review]
LGTM

**Comment by NejcZdovc** [comment]
CI failed on not related tests

**Comment by bbondy** [comment]
Merging for Nejc since he asked me to force merge on Slack

**Comment by kjozwiak** [comment]
Verification **PASSED** on `macOS 10.15.7 x64` using the following build:

```
Brave | 1.18.10 Chromium: 86.0.4240.111 (Official Build) nightly (x86_64)
-- | --
Revision | b8c36128a06ebad76af51591bfec980224db5522-refs/branch-heads/4240@{#1290}
OS | macOS Version 10.15.7 (Build 19H2)
```

* Verified the STR/Cases outlined via https://github.com/brave/brave-browser/issues/12274#issue-727360814 & https://github.com/brave/brave-core/pull/6939#issue-509016349.

**`Restore with Balance`** | **`Restore without Balance`**
----------------------------|------------------------------
Screen Shot 2020-10-25 at 11 21 38 AM | Screen Shot 2020-10-25 at 11 25 16 AM

* ensured that you can `Copy`, `Print` & `Save as File`

**Pull request merged**